### PR TITLE
Kiera - Hotel - Octos

### PR DESCRIPTION
**Pull request by Krashaune**
# Hotel
Congratulations! You're submitting your assignment!

## Comprehension Questions
Question | Answer
:------------- | :-------------
Describe a design decision you had to make when working on this project. What options were you considering? What helped you make your final decision? | In the early stages of the project i thought I needed a room class to store the rooms and such but, a room would never be instantiated soon I realized the room would have no behaviors so when I went to code there was no need for it and the behaviors would primarily be on the hotel class.
Describe a concept that you gained more clarity on as you worked on this assignment. | Test driven development how it really drives the code you write. When I would try to code without creating a test and watching it fail I would write nonsense code and or get confused about what the method was supposed to do.
Describe a _nominal test_ that you wrote for this assignment. |  a nominal test i wrote was create multiple reservation which tested to see if when multiple reservations were made that the reservations array contained them all
Describe an _edge case test_ that you wrote for this assignment. | create reservations when dates overlap which tested to make sure that when multiple reservations made the same dates their room numbers were not the same.
How do you feel you did in writing pseudocode first, then writing the tests and then the code? |I think I did pretty well with writing pseudocode first. As mentioned in above i did veer away from tests then code but, was reminded why it was best. My pseudocode is what kept me on track and helped with determining what to right for tests. I love pseudocoding!

### CODE REVIEW

File: lib/hotel.rb [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+  # find an available room for the new reservation
+  # assign the room number to the new reservation
+  def find_available_room
+    if reservations.length == 0
+      room_number = @rooms.sample
+    else
+      # go through @rooms array and sample a room_number
+      # with that room_number call reserevation_overlap?
+      # determine whether the new reservations date range overlaps
+      # with reservations in the reservation array
```
Comment (kariabancroft) [inline lib/hotel.rb]: Doesn't quite seem like the code below accomplished everything you've outlined here

File: lib/hotel.rb [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    start_date = reservation.start_date
+    end_date = reservation.end_date
+    room_number = find_available_room
+    new_res = Reservation.new(start_date, end_date,room_number)
+    reservations << new_res
+    return new_res
+  end
+
+  # overlap? reservation needs to take in a date range and determine if it overlaps with self any other reservations ie boolean output
+  def reservation_overlap?(reservation)
```
Comment (kariabancroft) [inline lib/hotel.rb]: Methods that have the `?` should always return true or false. Here you're only returning one of those

**Comment by kariabancroft** [comment]
# Hotel
## What We're Looking For

Feature | Feedback
---     | ---
**Design** |
Demonstrated classes having a single responsibility | Yes
Demonstrated loose coupling | Yes
Methods demonstrate a good use of encapsulation, inputs and outputs | Yes
Wave 1 requirements |Yes
Wave 2 requirements | Incomplete
Wave 3 requirements |No

You did a good job in several of these method by outlining the purpose and then completing the implementation. I'd like to see how you continued to design given the completion of the remainder of the requirements.
